x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/index/IndexResolution.java

```diff
@@ -56,18 +57,22 @@ public static IndexResolution notFound(String name) {
 
     // all indices found by field-caps
     private final Set<String> resolvedIndices;
+    @Nullable
+    private final FieldCapabilitiesFailure localResolutionFailure;
```

Comment:
Can we merge this with `unavailableClusters`?

Reply:
I do not think so. I just updated the implementation to look for unavailable shards rather than generic failures and I would like to keep `unavailableClusters` remote only as documented:
https://github.com/elastic/elasticsearch/blob/d1853569bfb20a81f37be4281f2686e3b6f6a837/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/index/IndexResolution.java#L63